allennlp/modules/token_embedders/bert_token_embedder.py

```diff
@@ -0,0 +1,111 @@
+"""
+A ``TokenEmbedder`` which uses one of the BERT models
+(https://github.com/google-research/bert)
+to produce embeddings.
+
+At its core it uses Hugging Face's PyTorch implementation
+(https://github.com/huggingface/pytorch-pretrained-BERT),
+so thanks to them!
+"""
+import logging
+
+import torch
+
+from pytorch_pretrained_bert.modeling import BertModel
+
+from allennlp.modules.token_embedders.token_embedder import TokenEmbedder
+from allennlp.nn import util
+
+logger = logging.getLogger(__name__)
+
+
+class BertEmbedder(TokenEmbedder):
+    """
+    A ``TokenEmbedder`` that produces BERT embeddings for your tokens.
+    Should be paired with a ``BertIndexer``, which produces wordpiece ids.
+
+    Most likely you probably want to use ``PretrainedBertEmbedder``
+    for one of the named pretrained models, not this base class.
+
+    Parameters
+    ----------
+    bert_model: ``BertModel``
+        The BERT model being wrapped.
+    """
+    def __init__(self,
+                 bert_model: BertModel) -> None:
+        super().__init__()
+        self.bert_model = bert_model
+        self.output_dim = bert_model.config.hidden_size
+
+    def get_output_dim(self) -> int:
+        return self.output_dim
+
+    def forward(self,
+                input_ids: torch.LongTensor,
+                offsets: torch.LongTensor = None,
+                token_type_ids: torch.LongTensor = None) -> torch.Tensor:
+        """
+        Parameters
+        ----------
+        input_ids: ``torch.LongTensor``
+            The wordpiece ids for each input sentence.
+        offsets: ``torch.LongTensor``, optional (default = None)
+            The BERT embeddings are one per wordpiece. However it's possible/likely
+            you might want one per original token. In that case, ``offsets``
+            should represent the indices of the last wordpiece for each original token.
+
+            That is, if you had the sentence "Definitely not", and if the corresponding
+            wordpieces were ["Def", "##in", "##ite", "##ly", "not"], then the input_ids
+            would be 5 wordpiece ids, and the offsets would be [3, 4]. If offsets are
+            provided, the returned tensor will contain the _last_ wordpiece embedding
+            for each token, and (in particular) will contain one embedding per token.
+            If offsets are not provided, the entire tensor of wordpiece embeddings
+            will be returned.
+        token_type_ids: ``torch.LongTensor``, optional (default = None)
+            If an input consists of two sentences (as in the BERT paper),
+            tokens from the first sentence should have type 0 and tokens from
+            the second sentence should have type 1.  If you don't provide this
+            (the default BertIndexer doesn't) then it's assumed to be all 0s.
```

Comment:
Is there a way to modify the indexer to provide this?  It's fine to do it in another PR, but I'd at least open an issue to track adding that.  Seems pretty important if you want to use this for SQuAD.